Sources/Datadog/RUM/RUMMonitor/Scopes/RUMUserActionScope.swift

```diff
@@ -120,11 +120,11 @@ internal class RUMUserActionScope: RUMScope, RUMContextProvider {
             dd: .init(),
             action: .init(
                 crash: nil,
-                error: .init(count: errorsCount.toInt64),
+                error: .init(count: max(0, errorsCount).toInt64),
                 id: actionUUID.toRUMDataFormat,
                 loadingTime: completionTime.timeIntervalSince(actionStartTime).toInt64Nanoseconds,
                 longTask: nil,
-                resource: .init(count: resourcesCount.toInt64),
+                resource: .init(count: max(0, resourcesCount).toInt64),
```

Comment:
Yes, those are sanity checks. It's not the case anymore in `RUMViewScope`, but here counts are eagerly increased and eventually decreased (dropped events), a decreased count will potentially wrap around when inconsistently going below 0. I should align with `RUMViewScope` and only do increases to get rid of this workaround.

Reply:
I see, indeed aligning it with only-increments model as in `RUMViewScope` would work better. It will also better utilise the mapping completion command and its `change` value.